### ISSUE TITLE
Inconsistent GUID(UUID) rendering

### ISSUE THREAD

**Issue by gongdo** [ChilliCream/hotchocolate#1234]
**Problem**
When you define two Guid fields by `IdType` and `UuidType`, The string representation of two fields will not be same.

**Code**
```csharp
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using HotChocolate;
using HotChocolate.Execution;
using HotChocolate.Types;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace hc_sample
{
    [TestClass]
    public class UnitTest1
    {
        public class Foo
        {
            public Guid Id { get; set; }
            public Guid OtherId { get; set; }
        }
        public class FooType : ObjectType<Foo>
        {
            protected override void Configure(IObjectTypeDescriptor<Foo> descriptor)
            {
                descriptor.Field(t => t.Id).Type<NonNullType<IdType>>();
                descriptor.Field(t => t.OtherId).Type<NonNullType<UuidType>>();
            }
        }
        private static readonly IEnumerable<Foo> fooes = 
            Enumerable
                .Range(0, 10)
                .Select(_ => new Foo
                {
                    Id = Guid.NewGuid(),
                    OtherId = Guid.NewGuid()
                })
                .ToList();
        public class Query
        {
            public Foo Foo(Guid id)
                => fooes.FirstOrDefault(o => o.Id == id);
            public IEnumerable<Foo> Fooes() => fooes;
        }
        public class QueryType : ObjectType<Query>
        {
        }

        [TestMethod]
        public async Task TestMethod1()
        {
            var schema = SchemaBuilder.New()
                .AddType<FooType>()
                .AddQueryType<QueryType>()
                .Create();
            
            var foo = fooes.First();
            var id = foo.Id;
            var query = @"
            {
                foo(id: """ + id.ToString() + @""") {
                    id
                    otherId
                }
            }";
            var result = await schema
                .MakeExecutable()
                .ExecuteAsync(query);
            
            Console.WriteLine(result.ToJson(true));
        }
    }
}
```

**Output Result**
```json
{
  "data": {
    "foo": {
      "id": "d0839794346a4bc0ba9d42488662e76b",
      "otherId": "fca35e17-9c0a-4989-8f90-d9e8114ddfdd"
    }
  }
}
```

**Expected Result**
```json
{
  "data": {
    "foo": {
      "id": "d0839794346a4bc0ba9d42488662e76b",
      "otherId": "fca35e179c0a49898f90d9e8114ddfdd"
    }
  }
}
```

Almost every code in HotChocolate uses `{Guid}.ToString("N")`, except for a field that defined by `UuidType`.
**Is this an expected behavior?**

I found that this behavior leads to inconsistent JSON result for an application. In .NET application it might not be a problem, because .NET can parses both type of Uuid string format. But the other languages, who knows?

By the way, I understand that the 'N' formatting reduces size, but I believe any UUID `type` should be rendered with `-` which is its canonical textual representation.
So the actual result that I want is:
```json
{
  "data": {
    "foo": {
      "id": "d0839794-346a-4bc0-ba9d-42488662e76b",
      "otherId": "fca35e17-9c0a-4989-8f90-d9e8114ddfdd"
    }
  }
}
```

Of course there's an work around; separate actual data type and graph type like this:
```csharp
        public class Foo
        {
            public Guid Id { get; set; }
            public Guid OtherId { get; set; }
        }
        public class FooGraph
        {
            public string Id { get; set; }
            public string OtherId { get; set; }
        }
```
Then resolves or converts the source some how(which is very confusing with HotChocolate fluent definition).

**Comment by michaelstaib** [comment]
In this case it is actually not inconsistent :) The IDType is as string serialisable type. So everything that goes in here becomes a sting. This means in case of the Guid it will become the ToString representation. 

In case of the Uuid we decided to have a short string with the `N` option.

But I see you complaint and I can understand.

For the first case you can control how the Guid is converted to a string through the TypeConversion API. 

The latter scalar does not yet allow customisation. 

So, I think let's make this configurable. We will allow the Uuid to serialise in a configured way. I think this way you get your dashes and we do not break the current behaviour.

**Comment by gongdo** [comment]
@michaelstaib I understand that the 'N' option behaviour is the default choice.
For now, as a workaround, I'm doing like this:

```
    public class GuidStringMiddleware
    {
        private readonly FieldDelegate next;

        public GuidStringMiddleware(FieldDelegate next)
        {
            this.next = next ?? throw new ArgumentNullException(nameof(next));
        }

        public virtual async Task InvokeAsync(IMiddlewareContext context)
        {
            await next(context);
            if ((context.Field.Member is PropertyInfo property
                    && property.PropertyType == typeof(Guid))
                || (context.Field.Member is FieldInfo field
                    && field.FieldType == typeof(Guid)))
            {
                context.Result = context.Result is Guid guid
                    ? guid.ToString()
                    : Guid.TryParse(context.Result.ToString(), out Guid guid1)
                        ? guid1.ToString()
                        : throw new InvalidOperationException($"Cannot parse the field result to Guid.");
            }
        }
    }

    public static class GuidStringMiddlewareExtensions
    {
        public static IObjectFieldDescriptor UseGuidString(
            this IObjectFieldDescriptor descriptor)
        {
            return descriptor
                .Use<GuidStringMiddleware>();
        }

        public static ISchemaBuilder UseGuidString(
            this ISchemaBuilder builder)
        {
            return builder
                .Use<GuidStringMiddleware>();
        }
    }
```

And add the middleware to the schema when I want to replace the current behaviour:
```
SchemaBuilder.New()
    .UseGuidString()
    // ...
    .Create();
```
Or add the middleware to the field definition only:
```
descriptor
    .Field(t => t.Id)
    .UseGuidString()
    .Type<NonNullType<IdType>>();
```

It's quite ugly code(double conversion; Guid-string-Guid), but it satisfies my requirement. ;)

I tried to find appropriate method, but I think it requires way more work than my quick work-around.
It would be great if there's some official way to configure default string serialization.

**Comment by michaelstaib** [comment]
I will move this to version 11 since we did a lot of changes to the scalars there. Adding this feature to 10 would result in to many merge conflicts.

**Comment by michaelstaib** [comment]
#1301

**Comment by michaelstaib** [comment]
This one is now implemented and will be included into 11.0.0-preview.69.
